tools/klee-zesti/klee-zesti

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+import sys
+import os
+import subprocess
+import tempfile
+import select
+HELP="""OVERVIEW: ZESTI like wrapper of KLEE
+
+USAGE:  klee-zesti [klee-options] <input bytecode> <concrete program arguments>
+
+WARNING this script is not eqvivalent to ZESTI in ICSE 2012. It just provides a similar interface to klee. Namely it first explores the path of <concrete prgoram arguments> and then continues symbolic execution from that point. Most importantly it does not implement the ZESTI searcher.
+"""
+
+
+#GEN_BOUT="/data/klee/build7.0/bin/gen-bout"
+#KLEE="/data/klee/build7.0/bin/klee"
+KLEE="klee"
+GEN_BOUT="gen-bout"
+
+def find_klee_bin_dir():
+  global KLEE
+  global GEN_BOUT
+  bin_dir = os.path.dirname(os.path.realpath(__file__))
+  KLEE = bin_dir + "/klee"
+  GEN_BOUT = bin_dir + "/gen-bout"
+  print("using", KLEE)
+  if not os.path.isfile(KLEE):
+      print("WARNING can't find klee at " + KLEE)
+      print("Using klee in PATH")
+      KLEE="klee"
```

Comment:
```
... = shutil.which('klee')
print(...)
```
Is more helpful. One could even check cwd... or add an env variable (`ZESTI_INTERPRETER`) :smile:.

Reply:
But using PATH gives the same functionality with 0 lines of code?

Why is the`shutil` more helpful?

Reply:
1. you might have multiple klee's in your path
2. you can fail in case none is found
3. you can print the full path